### PR TITLE
Template selection menu fix

### PR DESCRIPTION
**Pull request by juliamourac**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
If the last item of a page that is not the first is deleted, the user can still see the page showing "no templates on this page".

* **What is the new behavior (if this is a feature change)?**
After the last template of a page is deleted the pagination goes to the latest remaining page.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#1179

* **Other information**:

### CODE REVIEW

File: src/js/views/templates/TemplateList/Sidebar/index.js [hunk 1]
```diff
@@ -372,25 +372,30 @@ class Sidebar extends Component {
 
     deleteTemplate() {
         const { template } = this.state;
-        const { temp_opex, toogleSidebar, t } = this.props;
+        const {
+ temp_opex, toogleSidebar, t, numOfTempPage,
```
Comment (nunesjoab) [inline src/js/views/templates/TemplateList/Sidebar/index.js]: Expected indentation of 12 spaces, found 0. In object destructuring with 3 or more variables, each of these variables should be in a new line.

File: src/js/views/templates/TemplateOperations.js [hunk 2]
```diff
@@ -23,13 +23,13 @@ class TemplateOperations extends GenericOperations {
...
     whenRemoveItemFromLastPage() {
-        if (this.paginationParams.page_num > 0) {
+        if (this.paginationParams.page_num > 1) {
             this.paginationParams.page_num = this.paginationParams.page_num - 1;
         }
     }
 
     _fetch() {
-        const res = Object.assign({}, this.paginationParams, this.filterParams);
+        const res = { ...this.paginationParams, ...this.filterParams };
```
Comment (nunesjoab) [inline src/js/views/templates/TemplateOperations.js]: Nice!

File: src/js/views/devices/Actions/DeviceHandlerActions.js [hunk 3]
```diff
@@ -95,7 +95,7 @@ class DeviceHandlerActions {
         });
     }
 
-    triggerRemoval(device, cb) {
+    triggerRemoval(device, n, cb) {
```
Comment (mprevide) [inline src/js/views/devices/Actions/DeviceHandlerActions.js]: Remove unused param

Comment (juliamourac) [reply src/js/views/devices/Actions/DeviceHandlerActions.js]: I put it there so the call made in remove() method in Devices.js directs towards this function.

File: src/js/views/devices/Sidebar/index.js [hunk 4]
```diff
@@ -308,9 +308,13 @@ class Sidebar extends Component {
 
     remove() {
         const { device } = this.state;
-        const { ops, t } = this.props;
-        FormActions.triggerRemoval(device, () => {
+        const { ops, t, numOfDevPage } = this.props;
+
+        FormActions.triggerRemoval(device, numOfDevPage, () => {
             toaster.success(t('devices:alerts.remove'));
+            if (numOfDevPage === 1) {
```
Comment (mprevide) [inline src/js/views/devices/Sidebar/index.js]: I think the right way It's you use numOfDevPage in state or pass numOfDevPage as parameter.

**Pull request merged**